src/catch2/matchers/catch_matchers_elements_are.hpp

```diff
@@ -8,61 +8,124 @@
+        class UnorderedElementsAreElementsMatcher final
+            : public MatcherGenericBase {
+            TargetRangeLike m_desired;
+            Equality m_predicate;
+
+        public:
+            template <typename TargetRangeLike2, typename Equality2>
+            UnorderedElementsAreElementsMatcher( TargetRangeLike2&& range,
+                                                 Equality2&& predicate ):
+                m_desired( CATCH_FORWARD( range ) ),
+                m_predicate( CATCH_FORWARD( predicate ) ) {}
+
+            template <typename RangeLike> bool match( RangeLike&& rng ) const {
+                return std::is_permutation( m_desired.begin(),
+                                            m_desired.end(),
+                                            rng.begin(),
+                                            rng.end(),
+                                            m_predicate );
+            }
+
+            std::string describe() const override {
+                return "unordered elements are " +
+                       ::Catch::Detail::stringify( m_desired );
+            }
+        };
+
+        /**
+         * Creates a matcher that checks if all elements in a range are equal
+         * to all elements in another range.
+         *
+         * Uses `std::equal_to` to do the comparison
+         */
+        template <typename RangeLike>
+        std::enable_if_t<!Detail::is_matcher<RangeLike>::value,
+                         ElementsAreElementsMatcher<RangeLike, std::equal_to<>>>
+        ElementsAre( RangeLike range ) {
+            return { CATCH_FORWARD( range ), std::equal_to<>{} };
+        }
+
+        /**
+         * Creates a matcher that checks if all elements in a range are equal
+         * to all elements in another range.
+         *
+         * Uses to provided predicate `eq` to do the comparisons
+         */
+        template <typename RangeLike, typename Equality>
+        ElementsAreElementsMatcher<RangeLike, Equality>
+        ElementsAre( RangeLike&& range, Equality&& predicate ) {
+            return { CATCH_FORWARD( range ), CATCH_FORWARD( predicate ) };
+        }
+
+        /**
+         * Creates a matcher that checks if all elements in a range are equal
+         * to all elements in another range, in some permutation
+         *
+         * Uses `std::equal_to` to do the comparison
+         */
+        template <typename RangeLike>
+        std::enable_if_t<
+            !Detail::is_matcher<RangeLike>::value,
+            UnorderedElementsAreElementsMatcher<RangeLike, std::equal_to<>>>
+        UnorderedElementsAre( RangeLike range ) {
+            return { CATCH_FORWARD( range ), std::equal_to<>{} };
+        }
+
+        /**
+         * Creates a matcher that checks if all elements in a range are equal
+         * to all elements in another range, in some permuation.
+         *
+         * Uses to provided predicate `eq` to do the comparisons
```

Comment:
nitpick: `eq` is not the name of the arg in this function

Reply:
Already fixed in followup: 7c09700